### PR TITLE
Extra tests for 04553

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Re-upload of #2011 (deferred) 

Pleasant surprise that my original issue was fixed in #2182 and these extra test pass. They test if the descriptor types are `VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE` and `VK_DESCRIPTOR_TYPE_SAMPLER`

### CODE REVIEW

**Comment by ncesario-lunarg** [comment]
Running through [internal CI](http://erusea:4040/8047).

File: tests/vklayertests_pipeline_shader.cpp [hunk 1]
```diff
@@ -9306,3 +9306,204 @@ TEST_F(VkLayerTest, InvalidPrimitiveFragmentShadingRateWriteMultiViewportLimit)
...
+    pipeline_unused.SetViewport(m_viewports);
+    pipeline_unused.SetScissor(m_scissors);
+    pipeline_unused.AddShader(&vs);
+    VkPipelineObj pipeline_function(m_device);
+    pipeline_function.AddDefaultColorAttachment();
+    pipeline_function.SetViewport(m_viewports);
+    pipeline_function.SetScissor(m_scissors);
+    pipeline_function.AddShader(&vs);
+
+    // 3 different pipelines for 3 different shaders
```
Comment (ncesario-lunarg) [inline tests/vklayertests_pipeline_shader.cpp:9417]: 4 different pipelines/shaders?

**Review by ncesario-lunarg — APPROVED** [review]
LGTM, thanks @sfricke-samsung! Will merge once internal CI finishes.

File: tests/vklayertests_pipeline_shader.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply tests/vklayertests_pipeline_shader.cpp:9417]: I think I ment "3 bad and 1 good" ... will quick clean that up in a bit

**Pull request merged**